test/types/document.test.ts

```diff
@@ -164,4 +167,22 @@ function gh11435() {
   ItemSchema.pre('validate', function preValidate() {
     expectType<Model<unknown>>(this.model('Item1'));
   });
+}
+
+function m0_0aDocument() {
+  const AutoTypedModel = autoTypedModel();
+  const AutoTypeModelInstance = new AutoTypedModel({ unExistProperty: 1, description: 2 });
+
+  expectType<M0_0aAutoTypedSchemaType['schema']['userName']>(AutoTypeModelInstance.userName);
+  expectType<M0_0aAutoTypedSchemaType['schema']['favoritDrink']>(AutoTypeModelInstance.favoritDrink);
+  expectType<M0_0aAutoTypedSchemaType['schema']['favoritColorMode']>(AutoTypeModelInstance.favoritColorMode);
+  expectType<number>(AutoTypeModelInstance.unExistProperty);
+  expectType<number>(AutoTypeModelInstance.description);
+
+  /* -------------------------------------------------------------------------- */
```

Comment:
nitpick
I am not fan of "blocks" like these
Just simple
// Document-Method-tests 

Also the other "blocks" should be changed...